client/src/main/java/io/split/engine/matchers/Semver.java

```diff
@@ -0,0 +1,178 @@
+import java.util.Arrays;
+
+public class Semver {
+    private final String MetadataDelimiter = "+";
+    private final String PreReleaseDelimiter = "-";
+    private final String ValueDelimiter = "\\.";
+
+    private static final Logger _log = LoggerFactory.getLogger(SplitParser.class);
+
+    private Long _major;
+    private Long _minor;
+    private Long _patch;
+    private String[] _preRelease;
+    private boolean _isStable;
+    private String _metadata;
+    private String _version;
+
+    public static Semver build(String version) {
+        if (version.isEmpty()) return null;
+        try {
+            return new Semver(version);
+        } catch (Exception ex) {
+            _log.error("An error occurred during the creation of a Semver instance:", ex.getMessage());
+            return null;
+        }
+    }
+
+    public String Version() {
+      return _version;
+    }
+
+    public Long Major() {
+        return _major;
+    }
+
+    public Long Minor() {
+        return _minor;
+    }
+
+    public Long Patch() {
+        return _patch;
+    }
+
+    public String[] PreRelease() {
+        return _preRelease;
+    }
+
+    public String Metadata() {
+        return _metadata;
+    }
+
+    public boolean IsStable() {
+        return _isStable;
+    }
+
+    /**
+     * Precedence comparision between 2 Semver objects.
+     *
+     * @return the value {@code 0} if {@code this == toCompare};
+     *         a value less than {@code 0} if {@code this < toCompare}; and
+     *         a value greater than {@code 0} if {@code this > toCompare}
+     */
+    public int Compare(Semver toCompare) {
+        if (_version.equals(toCompare.Version())) {
+            return 0;
+        }
+        // Compare major, minor, and patch versions numerically
+        int result = Long.compare(_major, toCompare.Major());
+        if (result != 0) {
+            return result;
+        }
+        result = Long.compare(_minor, toCompare.Minor());
+        if (result != 0) {
+            return result;
+        }
+        result = Long.compare(_patch, toCompare.Patch());
+        if (result != 0) {
+            return result;
+        }
+        if (!_isStable && toCompare.IsStable()) {
+            return -1;
+        } else if (_isStable && !toCompare.IsStable()) {
+            return 1;
+        }
+        // Compare pre-release versions lexically
+        int minLength = Math.min(_preRelease.length, toCompare.PreRelease().length);
+        for (int i = 0; i < minLength; i++) {
+            if (_preRelease[i].equals(toCompare.PreRelease()[i])) {
+                continue;
+            }
+            if ( Ints.tryParse(_preRelease[i]) != null &&  Ints.tryParse(toCompare._preRelease[i]) != null) {
+                return Integer.compare(Integer.parseInt(_preRelease[i]), Integer.parseInt(toCompare._preRelease[i]));
+            }
+            return AdjustNumber(_preRelease[i].compareTo(toCompare._preRelease[i]));
```

Comment:
why do you need the AdjustNumber?

Reply:
The compare function returns numbers higher than 1, so I need to adjust it to return -1, 0 or 1